### PR TITLE
feat: removed password error for safe password 

### PR DESCRIPTION
**Pull request by hansieodendaal**
- Removed PasswordError and opted to use `String` for the error type in `impl FromStr for SafePassword`. This was needed to ensure `SafePassword` can be used with `serde` and `clap`.
- Added a unit test to ensure `serde` and `clap` will work properly.

### CODE REVIEW

File: rustfmt.toml [hunk 1]
```diff
@@ -8,7 +8,7 @@ imports_layout = "HorizontalVertical"
 imports_granularity = "Crate"
 match_block_trailing_comma = true
 max_width = 120
-newline_style = "Unix"
```
Comment (stringhandler) [inline rustfmt.toml:11]: I don't think this is correct.... we will get different formats on different machines

File: src/password.rs [hunk 2]
```diff
@@ -68,15 +69,9 @@ impl SafePassword {
 }
 
 /// An error for parsing a password from a string
-#[derive(Debug)]
+#[derive(Debug, Snafu)]
 pub struct PasswordError;
```
Comment (stringhandler) [inline src/password.rs:73]: I would prefer this to be an enum with one variant and a better description. Where is this used?

Comment (hansieodendaal) [reply src/password.rs:73]: It is intrinsically used within the `WalletConfig` and `Cli` structs

Comment (hansieodendaal) [reply src/password.rs]: `PasswordError` removed, thanks for suggesting use of `Infallible` instead.

**Review by stringhandler — APPROVED** [review]
Happy, please update the PR description though

**Pull request merged**